helix-core/src/main/java/org/apache/helix/task/assigner/ThreadCountBasedTaskAssigner.java

```diff
@@ -74,19 +75,16 @@ public Map<String, TaskAssignResult> assignTasks(Iterable<AssignableInstance> as
   public Map<String, TaskAssignResult> assignTasks(
       AssignableInstanceManager assignableInstanceManager, Collection<String> instances,
       Iterable<TaskConfig> tasks, String quotaType) {
-    Iterable<AssignableInstance> assignableInstances = new HashSet<>();
+    Set<AssignableInstance> assignableInstances = new HashSet<>();
     // Only add the AssignableInstances that are also in instances
     for (String instance : instances) {
-      ((HashSet<AssignableInstance>) assignableInstances)
-          .add(assignableInstanceManager.getAssignableInstance(instance));
+      assignableInstances.add(assignableInstanceManager.getAssignableInstance(instance));
     }
 
     if (tasks == null || !tasks.iterator().hasNext()) {
-      logger.warn("No task to assign!");
       return Collections.emptyMap();
     }
-    if (assignableInstances == null || !assignableInstances.iterator().hasNext()) {
-      logger.warn("No instance to assign!");
```

Comment:
If we remove this, how to determine all quota full or something?

Reply:
Yeah, that's the dilemma. I'm trying to move the check and log to the build up phase in `AssignableInstanceManager`. I think we don't have to print this for every task assignment.

Reply:
`handleAdditionalTaskAssignment` is called once per pipeline, isn't it? Is this log a problem?